cli/src/cmd/forge/selectors.rs

```diff
@@ -0,0 +1,94 @@
+use crate::{
+    cmd::forge::build::{CoreBuildArgs, ProjectPathsArgs},
+    opts::forge::CompilerArgs,
+    utils::FoundryPathExt,
+};
+use clap::Parser;
+use ethers::prelude::artifacts::output_selection::ContractOutputSelection;
+use foundry_common::{
+    compile,
+    selectors::{import_selectors, SelectorImportData},
+};
+
+#[derive(Debug, Clone, Parser)]
+pub enum SelectorsSubcommands {
+    /// Upload selectors to registry
+    #[clap(visible_alias = "up")]
+    Upload {
+        /// The name of the contract to upload selectors for.
+        #[clap(required_unless_present = "all")]
+        contract: Option<String>,
+
+        /// Upload selectors for all contracts in the project.
+        #[clap(long, required_unless_present = "contract")]
+        all: bool,
+
+        #[clap(flatten)]
+        project_paths: ProjectPathsArgs,
```

Comment:
Let's also document this field

Reply:
tbf I don't really know how to document this field: it doesn't settable by the user and it's not documented over on [fourbytes.rs](https://github.com/foundry-rs/foundry/blob/master/cli/src/cmd/forge/fourbyte.rs#L24-L25) either..
any suggestions?

Reply:
these will be flattened and will use the existing documentation